apps/src/lab2/views/dialogs/DialogManager.tsx

```diff
@@ -45,85 +37,39 @@ const DialogManager: React.FunctionComponent<DialogManagerProps> = ({
   children,
 }) => {
   const [openDialog, setOpenDialog] = useState<DialogType | null>(null);
-  const [dialogCallback, setDialogCallback] = useState<(() => void) | null>(
-    () => null
-  );
-  const [dialogTitle, setDialogTitle] = useState<string | undefined>(undefined);
-  const [dialogMessage, setDialogMessage] = useState<string | undefined>(
-    undefined
-  );
-  const [dialogConfirmText, setDialogConfirmText] = useState<
-    string | undefined
-  >(undefined);
+  const [dialogArgs, setDialogArgs] = useState<AnyDialogType>();
 
   const showDialog = useCallback(
-    (
-      dialogType: DialogType,
-      callback: () => void,
-      title?: string,
-      message?: string,
-      confirmText?: string
-    ) => {
-      setDialogTitle(title);
-      setDialogMessage(message);
-      setDialogConfirmText(confirmText);
-      setOpenDialog(dialogType);
-      setDialogCallback(() => callback);
+    ({type, ...dialogArgs}: TypedDialogProps) => {
+      setOpenDialog(type);
+      setDialogArgs(dialogArgs);
     },
-    [
-      setDialogTitle,
-      setDialogMessage,
-      setDialogConfirmText,
-      setOpenDialog,
-      setDialogCallback,
-    ]
+    [setDialogArgs]
   );
 
-  const handleConfirm = useCallback(() => {
-    if (dialogCallback) {
-      dialogCallback();
-      setOpenDialog(null);
-    }
-  }, [dialogCallback, setOpenDialog]);
-
-  const handleCancel = useCallback(() => {
-    setOpenDialog(null);
-  }, [setOpenDialog]);
+  const closeDialog = useCallback(() => setOpenDialog(null), [setOpenDialog]);
 
-  const DialogView = openDialog && dialogCallback && DialogViews[openDialog];
+  // Allow the any because if it's NOT any, then line 63 with DialogView's args will toss an error.
+  // Keep this until we have a better solution. ¯\_(ツ)_/¯
+  // The typing on the `showDialog` function ensures the props are correct, so we're still safe'
+  // eslint-disable-next-line
+  const DialogView: any = openDialog && dialogArgs && DialogViews[openDialog];
```

Comment:
If anybody can make typescript happy w/o the explicit `any` cast, that'd be keen. 😠 

It's still type safe because the `showDialog` function is so strict, but it feels icky.